lib/palettes.cpp

```diff
@@ -7703,16 +107,15 @@ void PaletteState::Init()
       MaxTextureSize = std::min(MaxTextureSize, 4096);
       {
          std::lock_guard<std::mutex> lk{init_mtx};
-         Init_Palettes();
       }
 
-      GLuint paletteTexIds[Num_RGB_Palettes][2];
+      vector<array<GLuint, 2>> paletteTexIds(Palettes->NumPalettes());
       GLuint alphaTexId;
 
-      glGenTextures(Num_RGB_Palettes * 2, &(paletteTexIds[0][0]));
+      glGenTextures(Palettes->NumPalettes() * 2, &(paletteTexIds[0][0]));
```

Comment:
I thought this will go `Texture`, i.e., it will hold the actual OpenGL texture 🤔 . It would also save some memory, because `texture_data` are needed only temporarily to prepare the textures 😉 .

Reply:
I also had this in mind but was holding off to wrap this up 😅.. but I agree with you that it would be cleaner. I'll make this change - some other things like global alpha texture (from `PaletteState::GenerateAlphaTexture`) will also need to be refactored similarly.

Reply:
See [4d4be89](https://github.com/GLVis/glvis/pull/289/commits/4d4be8913d2cfd56fc65f60359cf5bcb1f6c01d2)

Reply:
Cool, step in good direction 👍 A few things:

- You forgot about `first_init`
- CI complains about ordering of inits in constructor
- Get rid of `PaletteTexIds`, textures can generate the ids by themselves. Not batched, but that does not matter much, I believe.
- The statics of `Texture` should be private. It uses only `rgba_internal` from the formats, alphas do not belong there. WIP, I guess. 🚧 
- Methods `SetCycle/Color` should be private as they set the texture to a semi-invalid state without update of the size. I do not see any separate usage, so maybe you may merge them with `UpdateTextureSize()`? 🤔

Reply:
I think these are all addressed now, thanks!

Reply:
Looks good here 😉